### PR TITLE
Added documentation for base.WrappedException

### PR DESCRIPTION
**Pull request by booleanings**
PR related to #24043 

Tried to base it off the documentation guide and used `DimensionMismatch` exception doc as an example.

Could possibly add additional info about `LoadError` and `InitError` but don't think this will be necessary

### CODE REVIEW

File: base/boot.jl [hunk 1]
```diff
@@ -260,8 +260,12 @@ struct AssertionError <: Exception
 end
 AssertionError() = AssertionError("")
 
-#Generic wrapping of arbitrary exceptions
-#Subtypes should put the exception in an 'error' field
+"""
+    WrappedException([msg])
+
+wrap arbitrary exceptions generically.
```
Comment (mschauer) [inline base/boot.jl]: Generic type for `Exception`s wrapping another `Exception`, such as `LoadError` and `InitError`.
Those exceptions contain information about the the root cause of an exception. Subtypes define a field `error` containing the causing`Exception`.

**Comment by xorJane** [comment]
Thanks for the PR!

I think CI is failing because `boot.jl` is a bit finnicky. I encountered the same issue in #22521. Moving the docstring to `helpdb/Base.jl` should fix the issue! (You can check that it does by rebuilding Julia locally after making the change.)

If you need any help, let me know here or on slack. 😄

**Comment by JeffBezanson** [comment]
Looks like the new location for the doc string was left out of this somehow.

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -854,6 +854,15 @@ Generic error type. The error message, in the `.msg` field, may provide more spe
 ErrorException
 
 """
+    WrappedException(msg)
+
+Generic type for `Exception`s wrapping another `Exception`, such as `LoadError` and
+`InitError`. Those exceptions contain information about the the root cause of an
+exception. Subtypes define a field `error` containing the causing `Exception`.
+"""
+WrappedException
```
Comment (fredrikekre) [inline base/docs/basedocs.jl]: This should be `Core.WrappedException` and things will be alright :)

**Review by fredrikekre — APPROVED** [review]

**Review by Sacha0 — APPROVED** [review]
Thanks @aidaekay! :)

**Comment by xorJane** [comment]
👍

**Pull request merged**